src/plugins/intel_cpu/src/nodes/interaction.cpp

```diff
@@ -0,0 +1,229 @@
+    outputFeaturesLen = interactFeatureSize + featureSize;
+    std::vector<int64_t> lhsShape({static_cast<int64_t>(inputSizes), static_cast<int64_t>(featureSize)});
+    std::vector<int64_t> lhsStride({static_cast<int64_t>(featureSize), 1});
+    std::vector<int64_t> rhsShape({static_cast<int64_t>(featureSize), static_cast<int64_t>(inputSizes)});
+    std::vector<int64_t> rhsStride({1, static_cast<int64_t>(featureSize)});
+    std::vector<int64_t> resShape({static_cast<int64_t>(inputSizes), static_cast<int64_t>(inputSizes)});
+    std::vector<int64_t> resStride({static_cast<int64_t>(inputSizes), 1});
+    auto dataType = DnnlExtensionUtils::IEPrecisionToDataType(dataPrecision);
+    auto src_md = memory::desc(lhsShape, dataType, lhsStride);
+    auto weights_md = memory::desc(rhsShape, dataType, rhsStride);
+    auto dst_md = memory::desc(resShape, dataType, resStride);
+    auto matmul_d = matmul::desc(src_md, weights_md, dst_md);
+    primitive_attr matmul_attr;
+    auto matmul_pd = matmul::primitive_desc(matmul_d, matmul_attr, getEngine());
+    prim.reset(new matmul(matmul_pd));
+    featureSizes.resize(inputSizes, featureSize);
+    std::vector<InferenceEngine::TensorDesc> internalMemDesc = {
+        InferenceEngine::TensorDesc(
+            dataPrecision,
+            {inputSizes, featureSize},
+            InferenceEngine::Layout::HW),
+        InferenceEngine::TensorDesc(
+            dataPrecision,
+            {inputShapes.size(), inputShapes.size()},
+            InferenceEngine::Layout::HW),
+        InferenceEngine::TensorDesc(
+            dataPrecision,
+            {interactFeatureSize},
+            InferenceEngine::Layout::ANY)
+    };
+
+    if (dataPrecision == InferenceEngine::Precision::FP32) {
+        initializeInternalMemory<float>(internalMemDesc);
+    } else {
+        initializeInternalMemory<int16_t>(internalMemDesc);
+    }
+
+    inputMemPtr = std::make_shared<Memory>(getEngine());
+    outputMemPtr = std::make_shared<Memory>(getEngine());
+    auto inDesc = MemoryDescUtils::convertToDnnlBlockedMemoryDesc(inputPtr->getTensorDesc());
+    auto outDesc = MemoryDescUtils::convertToDnnlBlockedMemoryDesc(outputPtr->getTensorDesc());
+    inputMemPtr->Create(inDesc, inputPtr->buffer());
+    outputMemPtr->Create(outDesc, outputPtr->buffer());
+}
+
+void Interaction::executeDynamicImpl(dnnl::stream strm) {
+    execute(strm);
+}
+
+bool Interaction::isExecutable() const {
+    return true;
+}
+
+bool Interaction::isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op,
+        std::string& errorMessage) noexcept {
+    try {
+        const auto interaction = std::dynamic_pointer_cast<const InteractionNode>(op);
+        if (!interaction) {
+            errorMessage = "Only Interaction operation is supported";
+            return false;
+        }
+        auto elementType = interaction->get_output_type();
+        if (!one_of(elementType, ngraph::element::f32, ngraph::element::bf16)) {
```

Comment:
Don't need to validate precision here. Just set preferred precision in PortConfigurator in fn *initSupportedPrimitiveDescriptors*.

Reply:
remove to `initSupportedPrimitiveDescriptors`